### PR TITLE
Unification of repositories in CapsulaHub

### PR DESCRIPTION
**Pull request by dmitriyslaym**
https://github.com/capsulajs/capsulajs/issues/31

**Release notes:**
Prepared Lerna mono repo with common TS, lint, prettier, webpack, cypress configs, prepared templates for components and services packages, added documentations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,51 @@
...
+
+## Packages
+
+The list of all the packages that are included in the repository.
+
+### Core
+
+Core packages, that are used for the internal implementation of Capsulahub.
+
+1) workspace
```
Comment (MikeSquall) [inline README.md]: Could be good here (and for each package) to link the README of the package

File: packages/utils/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Utils
+
+Common utils for Exchange Sandbox and its extensions.
```
Comment (MikeSquall) [inline packages/utils/README.md]: Not for Exchange Sandbox

File: README.md [hunk 3]
```diff
@@ -1,2 +1,51 @@
-# capsulahub
-The main repository for CapsulaJS
+# Capsulahub
+The main repository of CapsulaJS.
```
Comment (MikeSquall) [inline README.md]:
```suggestion
An awesome toolkit to develop and test your micro-frontend services !
```

File: README.md (hunk 1 above)
Comment (dmitriyslaym) [reply README.md]: That's the plan, but only after we create these packages.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,52 @@
...
+## Packages
+
+The list of all the packages that are included in the repository.
+
+### Core
+
+Core packages, that are used for the internal implementation of Capsulahub.
+
+1) workspace
+2) rendered
```
Comment (stephanebenayoun) [inline README.md]:
```suggestion
2) renderer
```

And this service is not really internal, you need to specify it in your config like any other extension

File: README.md [hunk 5]
```diff
@@ -1,2 +1,52 @@
...
+
+1) cli
+
+### Service extensions
+
+The packages, that provide service extensions for Workspace (they should be included in **services**) prop in **Workspace** configuration.
+
+These extensions add some logic to an application, don't have a UI representation and have to register itself in **Workspace**, if they want to be available later within **Workspace**.
+
+1) service-configuration
```
Comment (stephanebenayoun) [inline README.md]: Will be good to have a table with all packages and short description (one line)

File: templates/service/README.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# Service Template extension
```
Comment (stephanebenayoun) [inline templates/service/README.md]: Missing :
- example of config to copy/paste in workspace config
- url of deployed for develop and master

File: templates/service/package.json [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "@capsulajs/capsula-hub-service-template",
```
Comment (stephanebenayoun) [inline templates/service/package.json]: didn't we agreed that we will always write capsulahub without dash ?

File: templates/widget/README.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
+# Template extension
```
Comment (stephanebenayoun) [inline templates/widget/README.md]: Missing :
- example of config to copy/paste in workspace config
- url of deployed for develop and master

File: templates/service/README.md (hunk 6 above)
Comment (dmitriyslaym) [reply templates/service/README.md]: url of the service inside capsulahub deployed for develop and master - what do you mean by this? Can you provide an example please?

File: lerna.json [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{
+  "npmClient": "yarn",
+  "useWorkspaces": true,
+  "packages": ["packages/*"],
+  "version": "0.0.1"
```
Comment (MikeSquall) [inline lerna.json]:
```suggestion
  "version": "independent"
```

**Pull request merged**